spec/asyncapi.md

```diff
@@ -529,167 +529,197 @@ defaultContentType: application/json
+        }
+      },
+      "parameters": {
+        "userId": {
+          "$ref": "#/components/parameters/userId"
+        }
+      },
+      "servers": ["rabbitmqInProd", "rabbitmqInStaging"],
+      "bindings": {
+        "amqp": {
+          "is": "queue",
+          "queue": {
+            "exclusive": true
           }
         }
-      }
-    }
-  },
-  "bindings": {
-    "amqp": {
-      "is": "queue",
-      "queue": {
-        "exclusive": true
+      },
+      "tags": [{
+        "name": "user",
+        "description": "User-related messages"
+      }],
+      "externalDocs": {
+        "description": "Find more info here",
+        "url": "https://example.com"
       }
     }
   }
 }
 ```
 
 ```yaml
-description: This channel is used to exchange messages about users signing up
-subscribe:
-  summary: A user signed up.
-  message:
-    description: A longer description of the message
-    payload:
-      type: object
-      properties:
-        user:
-          $ref: "#/components/schemas/user"
-        signup:
-          $ref: "#/components/schemas/signup"
-bindings:
-  amqp:
-    is: queue
-    queue:
-      exclusive: true
+channels:
+  userEvents:
+    address: 'users.{userId}'
+    addressDelimiter: '.'
+    description: This channel is used to exchange messages about user events.
+    messages:
+      userSignedUp:
+        $ref: '#/components/messages/userSignedUp'
+      userCompletedOrder:
+        $ref: '#/components/messages/userCompletedOrder'
+    parameters:
+      userId:
+        $ref: '#/components/parameters/userId'
+    servers:
+      - rabbitmqInProd
+      - rabbitmqInStaging
+    bindings:
+      amqp:
+        is: queue
+        queue:
+          exclusive: true
+    tags:
+      - name: user
+        description: User-related messages
+    externalDocs:
+      description: 'Find more info here'
+      url: 'https://example.com'
 ```
 
-Using `oneOf` to specify multiple messages per operation:
 
-```json
-{
-  "subscribe": {
-    "message": {
-      "oneOf": [
-        { "$ref": "#/components/messages/signup" },
-        { "$ref": "#/components/messages/login" }
-      ]
-    }
-  }
-}
-```
 
-```yaml
-subscribe:
-  message:
-    oneOf:
-      - $ref: '#/components/messages/signup'
-      - $ref: '#/components/messages/login'
-```
 
 
-Using explicit by-name references to the servers on which the channel is available:
+#### <a name="channelAddressExpressions"></a>Channel Address Expressions
+
+Channel addresses MAY contain expressions that can be used to define dynamic values.
+
+Expressions MUST be composed by a name enclosed in curly braces (`{` and `}`). E.g., `{userId}`.
+
+When the channel address contains [delimiters](#channelObjectAddressDelimiter), the expressions MUST be surrounded by either the delimiters, the beginning of the string, or the end of the string. E.g., assuming a `.` (dot) delimiter, `users.{userId}` is correct but `users.id{userId}` is not.
```

Comment:
This is covering most of the protocols supporting any form of wildcards: AMQP, MQTT, NATS, etc. It's still missing a Kafka feature that lets you subscribe to a regular expression. I tried to map it here but it's too complex and therefore left it out.

If you got any ideas on how to do it, I'm all ears. Otherwise, I'll open a follow-up issue and this can be included in future versions.

---

If you're curious, I came up with a solution like `users_{userId:/[\w\d]{8}/i}` but quickly got trapped in the [differences between regular expression engines](https://en.wikipedia.org/wiki/Comparison_of_regular_expression_engines) hole.

Kafka protocol doesn't define anything so it's hard for us to make a commitment here. Maybe go for the engine with fewer features, just to make sure everything is supported in every engine? Coming up with a subset ourselves? Should we just forget about it? Hot topic!